src/main/java/com/github/shyiko/mysql/binlog/event/RawBinaryLogEvent.java

```diff
@@ -0,0 +1,38 @@
+package com.github.shyiko.mysql.binlog.event;
+
+import com.github.shyiko.mysql.binlog.event.deserialization.BinaryLogEventDataReader;
+
+import java.nio.ByteBuffer;
+
+/**
+ * The class represents a fully read binary log event that hasn't been parsed yet. Events spanning multiple
+ * packets are still represented by a single {@link RawBinaryLogEvent}.
+ * <p>
+ * It can also represent an explicit EOF event sent by the server. Such an event doesn't have any data.
+ */
+public class RawBinaryLogEvent {
+    private final boolean isServerEof;
+    private final ByteBuffer buffer;
+
+
+    public BinaryLogEventDataReader getEventDataReader() {
+        return new BinaryLogEventDataReader(buffer.duplicate());
+    }
+
+    public boolean isServerEof() {
+        return isServerEof;
+    }
+
+    public static RawBinaryLogEvent createServerEof() {
```

Comment:
Looks like `createServerEof` is unused currently. Is that intended?

Reply:
Oh, you're right. I somehow forgot to add the usage.
Which means that I missed some unit tests. Will fix.

Reply:
Should be fixed now.